bndtools.core/src/bndtools/wizards/project/ImportBndWorkspaceWizard.java

```diff
@@ -0,0 +1,451 @@
+        monitor.subTask("Import Bnd project '" + bndProject.getName() + "'.");
+        final IWorkspace eclipseWorkspace = ResourcesPlugin.getWorkspace();
+        final IWorkspaceRoot eclipseWorkspaceRoot = eclipseWorkspace.getRoot();
+
+        deleteOldProjectFiles(Paths.get(bndWorkspace.getBaseURI()).resolve(bndProject.getName()));
+        // create generic project
+        final IProjectDescription projectDescription = eclipseWorkspace.newProjectDescription(bndProject.getName());
+        final IProject project = eclipseWorkspaceRoot.getProject(bndProject.getName());
+
+        IPath path = URIUtil.toPath(bndProject.getBaseURI());
+        if (Platform.getLocation().isPrefixOf(path)) {
+            projectDescription.setLocation(null);
+        } else {
+            projectDescription.setLocation(path);
+        }
+
+        project.create(projectDescription, monitor);
+        project.open(monitor);
+        project.setDefaultCharset("UTF-8", monitor);
+
+        setNatures(project, monitor, JavaCore.NATURE_ID, Plugin.BNDTOOLS_NATURE);
+
+        // create JavaProject
+        IJavaProject javaProject = JavaCore.create(project);
+
+        javaProject.open(monitor);
+        updateJavaProjectSettings(bndProject, javaProject, monitor);
+
+        importSourceAndOutputFolders(bndProject, project, javaProject, monitor);
+
+        // create BndToolsProject
+        // TODO why create BndProject?
+        BndProject bndtoolsProject = new BndProject(javaProject.getProject());
+        monitor.worked(1);
+    }
+
+    private void importSourceAndOutputFolders(Project bndProject, IProject workspaceProject, IJavaProject javaProject, IProgressMonitor monitor) throws Exception {
+        // remove defaults
+        removeClasspathDefaults(javaProject);
+
+        // Output
+        IFolder sourceOutput = workspaceProject.getFolder(URIUtil.toPath(bndProject.getSrcOutput().toURI()).makeRelativeTo(workspaceProject.getLocation()));
+        IPackageFragmentRoot outputRoot = javaProject.getPackageFragmentRoot(sourceOutput);
+
+        final boolean defaultOutputLocation = outputRoot.getElementName().equals(javaProject.getOutputLocation().makeRelativeTo(javaProject.getPath()).toString());
+
+        // Source (multiple possible)
+        for (File folder : bndProject.getSourcePath()) {
+            IFolder source = workspaceProject.getFolder(URIUtil.toPath(folder.toURI()).makeRelativeTo(workspaceProject.getLocation()));
+            // Now the created source folder should be added to the class entries of the project, otherwise compilation will fail
+            IPackageFragmentRoot root = javaProject.getPackageFragmentRoot(source);
+            List<IClasspathEntry> entries = new ArrayList<>(Arrays.asList(javaProject.getRawClasspath()));
+            entries.add(JavaCore.newSourceEntry(root.getPath(), null, defaultOutputLocation ? null : outputRoot.getPath()));
```

Comment:
But what if the project uses a maven-layout? Or do you mean that if the Bnd Workspace doesnt specify anything it has to be "bin" regardless what the Eclipse preference say?